jms/org.wso2.carbon.transport.jms/src/main/java/org/wso2/carbon/transport/jms/sender/JMSClientConnector.java

```diff
@@ -0,0 +1,175 @@
+            } else if (messageType.equals(JMSConstants.MAP_MESSAGE_TYPE) && carbonMessage instanceof MapCarbonMessage) {
+                message = session.createMapMessage();
+                MapMessage mapMessage = (MapMessage) message;
+                Enumeration<String> mapNames = ((MapCarbonMessage) carbonMessage).getMapNames();
+                while (mapNames.hasMoreElements()) {
+                    String key = mapNames.nextElement();
+                    mapMessage.setString(key, ((MapCarbonMessage) carbonMessage).getValue(key));
+                }
+            }
+
+            if (carbonMessage.getProperty(JMSConstants.PERSISTENCE) != null &&
+                carbonMessage.getProperty(JMSConstants.PERSISTENCE).equals(false)) {
+                messageProducer.setDeliveryMode(DeliveryMode.NON_PERSISTENT);
+            }
+
+            if (message != null) {
+                messageProducer.send(message);
+            }
+
+        } catch (JMSException e) {
+            throw new ClientConnectorException("Exception occurred while sending the message", e);
+        } finally {
+            try {
+                jmsConnectionFactory.closeMessageProducer(messageProducer);
+                jmsConnectionFactory.closeSession(session);
+                jmsConnectionFactory.closeConnection(connection);
+            } catch (JMSConnectorException e) {
+                throw new ClientConnectorException(e.getMessage(), e);
+            }
+        }
+        return false;
+    }
+
+    /**
+     * To create jms connection.
+     *
+     * @param propertySet Set of user defined properties
+     * @throws JMSConnectorException
+     * @throws JMSException
+     */
+    private void createConnection(Set<Map.Entry<String, String>> propertySet)
+            throws JMSConnectorException, JMSException {
+        Properties properties = new Properties();
+        for (Map.Entry<String, String> entry : propertySet) {
+            String mappedParameter = JMSConstants.MAPPING_PARAMETERS.get(entry.getKey());
+            if (mappedParameter != null) {
+                properties.put(mappedParameter, entry.getValue());
+            } else {
+                properties.put(entry.getKey(), entry.getValue());
+            }
+        }
+        JMSConnectionFactory jmsConnectionFactory;
+        if ((Integer.parseInt(properties.getProperty(JMSConstants.PARAM_CACHE_LEVEL)) > JMSConstants.CACHE_NONE) &&
+            this.jmsConnectionFactory != null) {
+            jmsConnectionFactory = this.jmsConnectionFactory;
+        } else {
+            jmsConnectionFactory = new CachedJMSConnectionFactory(properties);
+            this.jmsConnectionFactory = jmsConnectionFactory;
+        }
+
+        String conUsername = properties.getProperty(JMSConstants.CONNECTION_USERNAME);
+        String conPassword = properties.getProperty(JMSConstants.CONNECTION_PASSWORD);
+
+        Connection connection;
+        if (conUsername != null && conPassword != null) {
```

Comment:
These can be null since JMS connection can be created without username and password.